src/test/java/com/serenitydojo/fruitmarket/CatalogTest.java

```diff
@@ -1,30 +1,150 @@
 package com.serenitydojo.fruitmarket;
 
+import com.serenitydojo.*;
+import org.assertj.core.api.Assertions;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.serenitydojo.Fruit.*;
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class CatalogTest {
+    Catalog catalog;
+    @Before
+    public void addItemsToCatalog(){
+            catalog = Catalog.withItems(
+            new CatalogItem(Pear),
+            new CatalogItem(Apple),
+            new CatalogItem(Banana),
+            new CatalogItem(Orange)
+        );
+    }
 
     @Test
-    public void shouldBeAbleToUpdateTheCurrentPriceOfAFruit() {
-// TODO: Uncomment this code and make it work
-        // Catalog catalog = new Catalog();
-        // catalog.setPriceOf(Apple, 4.00);
-        // assertThat(catalog.getPriceOf(Apple)).isEqualTo(4.00);
+    public void shouldBeAbleToUpdateTheCurrentPriceOfAFruit() throws FruitUnavailableException {
+// TODO: Uncomment this code and make it work - Done
+  //       Catalog catalog = new Catalog();
+         catalog.setUnitPriceOf(Apple, 4.00);
+         assertThat(catalog.getUnitPriceOf(Apple)).isEqualTo(4.00);
     }
 
     @Test
     public void shouldListAvailableFruitsAlphabetically() {
-// TODO: Uncomment this code and make it work
-        // Catalog catalog = Catalog.withItems(
-        //         new CatalogItem(Pear, 1),
-        //         new CatalogItem(Apple, 1),
-        //         new CatalogItem(Banana, 1)
-        // );
-        // List<CatalogItem> availableFruits = catalog.getAvailableFruits();
-        // assertThat(availableFruits.get(0).getFruit()).isEqualTo(Apple);
-        // assertThat(availableFruits.get(1).getFruit()).isEqualTo(Banana);
-        // assertThat(availableFruits.get(2).getFruit()).isEqualTo(Pear);
+// TODO: Uncomment this code and make it work - Done
+         List<CatalogItem> availableFruits = catalog.getAvailableFruits();
+         System.out.println("Available fruits in alphabetical order: " + availableFruits);
+         assertThat(availableFruits.get(0).getFruit()).isEqualTo(Apple);
+         assertThat(availableFruits.get(1).getFruit()).isEqualTo(Banana);
+         assertThat(availableFruits.get(2).getFruit()).isEqualTo(Orange);
+         assertThat(availableFruits.get(3).getFruit()).isEqualTo(Pear);
+    }
+
+    @Test(expected = FruitUnavailableException.class )
+    public void shouldReportFruitPrice() throws FruitUnavailableException{
+        Assertions.assertThat(catalog.getUnitPriceOf(Apple)).isEqualTo(4.00);
+        Assertions.assertThat(catalog.getUnitPriceOf(Orange)).isEqualTo(5.50);
+        Assertions.assertThat(catalog.getUnitPriceOf(Banana)).isEqualTo(6.00);
+        Assertions.assertThat(catalog.getUnitPriceOf(Pear)).isEqualTo(4.50);
+        Assertions.assertThat(catalog.getUnitPriceOf(Grapefruit)).isEqualTo(4.50);
+    }
```

Comment:
this should not throw an exception, focus on price reporting and let the next test show the exception behavior
```suggestion
    public void shouldReportFruitPrice() throws FruitUnavailableException{
        Assertions.assertThat(catalog.getUnitPriceOf(Apple)).isEqualTo(4.00);
        Assertions.assertThat(catalog.getUnitPriceOf(Orange)).isEqualTo(5.50);
        Assertions.assertThat(catalog.getUnitPriceOf(Banana)).isEqualTo(6.00);
        Assertions.assertThat(catalog.getUnitPriceOf(Pear)).isEqualTo(4.50);
    }
```